snooty/parser.py

```diff
@@ -854,6 +862,180 @@ def handle_method_option(self, node: n.Directive) -> None:
+                        (
+                            composable_option
+                            for composable_option in composable_options
+                            if composable_option["value"] == option_key
+                        ),
+                        {},
+                    )
+                    if not option_from_spec or not composable_option:
+                        continue
+                    composable_option["selections"] = (
+                        composable_option["selections"] or []
+                    )
+                    if isinstance(composable_option["selections"], list):
+                        composable_option["selections"].append(
+                            {
+                                "value": option_from_spec.id,
+                                "text": option_from_spec.title,
+                            }
+                        )
+
+                default_values_found = default_values_found or all(
+                    child.selections[composable_id] == option_id
+                    for composable_id, option_id in (default_ids_dict.items())
+                )
+
+            except ChildValidationError:
+                continue
+
+        if not default_values_found:
+            self.diagnostics.append(
+                MissingChild(
+                    "composable-tutorial",
+                    "selected-content with selections {default_ids_as_string}",
+                    node.start[0],
+                )
+            )
+        node.composable_options = composable_options
+        node.options = {}
+
+    def handle_composable_content(
+        self,
+        node: n.ComposableContent,
+        spec_composables: List[Composable],
+    ) -> None:
+        selection_ids = re.split(r"\s*,\s*", node.options.get("selections", "").strip())
+        selections: Dict[str, str] = {}
+        # validate all selection ids
+        for idx in range(len(selection_ids)):
+            selection_id = selection_ids[idx]
+            spec_composable = spec_composables[idx]
+            allowed_selection_ids = list(map(lambda x: x.id, spec_composable.options))
+            # check if dependencies are met - then None is not allowed
+            met_dependencies: bool = all(
+                selections[composable_option_id] == option_value_id
+                for composable_option_id, option_value_id in (
+                    spec_composable.dependencies or {}
+                ).items()
+            )
+            if selection_id not in allowed_selection_ids and met_dependencies:
+                self.diagnostics.append(
+                    UnknownOptionId(
+                        "composable-tutorial",
+                        selection_id,
+                        allowed_selection_ids,
+                        node.start[0],
+                    )
+                )
+                break
+            composable_option_value = str(spec_composable.id)
```

Comment:
`spec_composable.id` seems to be of type string. Is this line necessary?

Reply:
removed!